src/main/java/org/folio/circulation/support/CqlSortBy.java

```diff
@@ -1,51 +1,48 @@
 package org.folio.circulation.support;
 
-public abstract class CqlSortBy {
-  public static CqlSortBy ascending(String index) {
-    return new AscendingCqlSortBy(index);
-  }
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
 
-  public static CqlSortBy descending(String index) {
-    return new DescendingCqlSortBy(index);
-  }
+import org.apache.commons.lang3.StringUtils;
 
-  public static CqlSortBy none() {
-    return new NoCqlSortBy();
-  }
-
-  protected abstract String applyTo(String query);
+public class CqlSortBy {
 
-  private static class DescendingCqlSortBy extends CqlSortBy {
+  private final List<CqlOrder> orders;
 
-    private final String index;
+  private CqlSortBy(List<CqlOrder> orders) {
+    this.orders = orders;
+  }
 
-    private DescendingCqlSortBy(String index) {
-      this.index = index;
-    }
+  public static CqlSortBy sortBy(CqlOrder... orders) {
+    return CqlSortBy.sortBy(Arrays.asList(orders));
+  }
 
-    @Override
-    protected String applyTo(String query) {
-      return String.format("%s sortBy %s/sort.descending ", query, index);
-    }
+  public static CqlSortBy sortBy(List<CqlOrder> orders) {
+    return new CqlSortBy(orders);
   }
 
-  private static class AscendingCqlSortBy extends CqlSortBy {
-    private final String index;
+  public static CqlSortBy ascending(String index) {
+    return CqlSortBy.sortBy(CqlOrder.asc(index));
+  }
 
-    AscendingCqlSortBy(String index) {
-      this.index = index;
-    }
+  public static CqlSortBy descending(String index) {
+    return CqlSortBy.sortBy(CqlOrder.desc(index));
+  }
 
-    @Override
-    protected String applyTo(String query) {
-      return String.format("%s sortBy %s/sort.ascending", query, index);
-    }
+  public static CqlSortBy none() {
+    return CqlSortBy.sortBy(Collections.emptyList());
```

Comment:
Why not use the private constructor here?

Reply:
Changed